src/cpp/shared/scripting/funcs.cpp

```diff
@@ -0,0 +1,987 @@
+
+
+extern "C" {
+#include "lua/src/lua.h"
+#include "lua/src/lualib.h"
+#include "lua/src/lauxlib.h"
+}
+
+#include "adventure/adv.h"
+#include "adventure/map.h"
+#include "combat/combat.h"
+#include "game/game.h"
+#include "gui/dialog.h"
+#include "town/town.h"
+#include "artifacts.h"
+
+#include "scripting/register.h"
+
+int StackIndexOfArg(int argNumber, int numArgs) {
+  return (numArgs - (argNumber - 1));
+}
+
+/************************************************ Deep binded objects ********************************************************/
+
+void MakeLuaHeroTable(lua_State *L, void *ptrAddr) {
```

Comment:
What about making a single func with an additional string parameter? "hero_mt" / "town_mt" will be passed there